src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -20,11 +21,10 @@ internal class SslStreamInternal
         private static readonly AsyncProtocolCallback s_resumeAsyncReadCallback = new AsyncProtocolCallback(ResumeAsyncReadCallback);
         private static readonly AsyncProtocolCallback s_readHeaderCallback = new AsyncProtocolCallback(ReadHeaderCallback);
         private static readonly AsyncProtocolCallback s_readFrameCallback = new AsyncProtocolCallback(ReadFrameCallback);
+        private static readonly Action<Task, object> s_freeWriteBufferCallback = new Action<Task, object>((_, o) => ArrayPool<byte>.Shared.Return((byte[])o));
```

Comment:
The continuation needs to call .GetAwaiter().GetResult() on the passed in antecedent task in order to propagate any exception that may have occurred.